core/src/main/scala/scalaz/Heap.scala

```diff
@@ -87,11 +87,15 @@ sealed abstract class Heap[A] {
 
   def toUnsortedList: List[A] = toUnsortedStream.toList
 
+  def toUnsortedIList: IList[A] = IList(toUnsortedStream: _*)
+
   def toStream: Stream[A] =
```

Comment:
I think it would be nice to the same approach as in `FingerTree`: use a `reduceTo` method to implement those `to*` methods. `reduceTo` would be:
```scala
  def reduceTo[M](implicit M: Reducer[A, M]): M = {
    @tailrec
    def go(acc: M, heap: Heap[A]): M = {
      heap.uncons match {
        case Some((a, as)) => go(M.cons(a, acc), as)
        case _ => acc
      }
    }
    go(M.zero, this)
  }
```

Reply:
but, thinking of it, `toStream` implementation should be kept as is to preserve laziness. `reduceTo` would be used only for strict containers.

Reply:
oups, `reduceTo` would produce in reverse order. So better leave it like that, sorry.

Reply:
I vaguely remember going through the same thought process here. 😅